test/compilable/test18295.d

```diff
@@ -0,0 +1,19 @@
+// REQUIRED_ARGS: -dip1000
+
+// https://issues.dlang.org/show_bug.cgi?id=18295
+
+// See test/fail_compilation/test18295.d for the non-`-dip1000` version`
+
+scope class C { int i; }    // Notice the use of `scope` here
+
+C increment(C c)            // Should not emit error here
```

Comment:
Yes it should error, you need `return scope` on the parameter for this to pass.

Reply:
Added `return scope` to the parameter